src/comp/comp.py

```diff
@@ -24,48 +24,49 @@ def __repr__(self):
 # Write a list comprehension that creates a list of names of everyone
 # whose name starts with 'D':
 print("Starts with D:")
-a = []
+a = [str(humanoid.name) for humanoid in humans if humanoid.name[0] == "D"]
 print(a)
 
 # Write a list comprehension that creates a list of names of everyone
 # whose name ends in "e".
 print("Ends with e:")
-b = []
+b = [str(humanoid.name) for humanoid in humans if list(humanoid.name)[-1] == "e"]
 print(b)
 
 # Write a list comprehension that creates a list of names of everyone
 # whose name starts with any letter between 'C' and 'G' inclusive.
 print("Starts between C and G, inclusive:")
-c = []
-print(c)
+letters = ['C', 'D', 'E', 'F', 'G']
+c = [str(humanoid.name) for humanoid in humans if humanoid.name[0] in letters]
+print(sorted(c))
 
 # Write a list comprehension that creates a list of all the ages plus 10.
 print("Ages plus 10:")
-d = []
+d = [int(humanoid.age + 10) for humanoid in humans]
 print(d)
 
 # Write a list comprehension that creates a list of strings which are the name
 # joined to the age with a hyphen, for example "David-31", for all humans.
 print("Name hyphen age:")
-e = []
-print(e)
+e = [str(f'{humanoid.name}-{humanoid.age}') for humanoid in humans]
+print(sorted(e))
 
 # Write a list comprehension that creates a list of tuples containing name and
 # age, for example ("David", 31), for everyone between the ages of 27 and 32,
 # inclusive.
 print("Names and ages between 27 and 32:")
-f = []
+f = [(humanoid.name, humanoid.age) for humanoid in humans if humanoid.age >= 27 and humanoid.age <=32]
 print(f)
 
 # Write a list comprehension that creates a list of new Humans like the old
 # list, except with all the names uppercase and the ages with 5 added to them.
 # The "humans" list should be unmodified.
-print("All names uppercase:")
-g = []
+print("All names uppercase, 5 added to age:")
+g = [Human((humanoid.name).upper(), (humanoid.age + 5)) for humanoid in humans]
 print(g)
 
 # Write a list comprehension that contains the square root of all the ages.
```

Comment:
Not much to say here, other than you nailed it. I don't have any constructive feedback since your code is really easy to read.